vnext/Microsoft.ReactNative/XamlLoadState.cpp

```diff
@@ -0,0 +1,178 @@
+#include "pch.h"
+#include "XamlLoadState.h"
+#include <sstream>
+
+XamlLoadState XamlLoadState::g_Instance;
+
+constexpr std::wstring_view wux{L"Windows.UI.Xaml.dll"};
+constexpr std::wstring_view mux{L"Microsoft.UI.Xaml.dll"};
+
+#pragma region NT APIs
+
+typedef LONG NTSTATUS;
+
+typedef struct _UNICODE_STRING {
+  USHORT Length;
+  USHORT MaximumLength;
+  PWSTR Buffer;
+} UNICODE_STRING, *PUNICODE_STRING;
+
+typedef UNICODE_STRING const *PCUNICODE_STRING;
+
+typedef struct _LDR_DLL_LOADED_NOTIFICATION_DATA {
+  ULONG Flags; // Reserved.
+  PCUNICODE_STRING FullDllName; // The full path name of the DLL module.
+  PCUNICODE_STRING BaseDllName; // The base file name of the DLL module.
+  PVOID DllBase; // A pointer to the base address for the DLL in memory.
+  ULONG SizeOfImage; // The size of the DLL image, in bytes.
+} LDR_DLL_LOADED_NOTIFICATION_DATA, *PLDR_DLL_LOADED_NOTIFICATION_DATA;
+
+typedef struct _LDR_DLL_UNLOADED_NOTIFICATION_DATA {
+  ULONG Flags; // Reserved.
+  PCUNICODE_STRING FullDllName; // The full path name of the DLL module.
+  PCUNICODE_STRING BaseDllName; // The base file name of the DLL module.
+  PVOID DllBase; // A pointer to the base address for the DLL in memory.
+  ULONG SizeOfImage; // The size of the DLL image, in bytes.
+} LDR_DLL_UNLOADED_NOTIFICATION_DATA, *PLDR_DLL_UNLOADED_NOTIFICATION_DATA;
+
+typedef union _LDR_DLL_NOTIFICATION_DATA {
+  LDR_DLL_LOADED_NOTIFICATION_DATA Loaded;
+  LDR_DLL_UNLOADED_NOTIFICATION_DATA Unloaded;
+} LDR_DLL_NOTIFICATION_DATA, *PLDR_DLL_NOTIFICATION_DATA;
+
+typedef LDR_DLL_NOTIFICATION_DATA const *PCLDR_DLL_NOTIFICATION_DATA;
+
+typedef VOID CALLBACK LDR_DLL_NOTIFICATION_FUNCTION(
+    _In_ ULONG NotificationReason,
+    _In_ PCLDR_DLL_NOTIFICATION_DATA NotificationData,
+    _In_opt_ PVOID Context);
+
+typedef LDR_DLL_NOTIFICATION_FUNCTION *PLDR_DLL_NOTIFICATION_FUNCTION;
+
+#define LDR_DLL_NOTIFICATION_REASON_LOADED 1
+
+NTSTATUS NTAPI LdrRegisterDllNotification(
+    _In_ ULONG Flags,
+    _In_ PLDR_DLL_NOTIFICATION_FUNCTION NotificationFunction,
+    _In_opt_ PVOID Context,
+    _Out_ PVOID *Cookie);
+
+NTSTATUS NTAPI LdrUnregisterDllNotification(_In_ PVOID Cookie);
+
+#pragma endregion
+
+LDR_DLL_NOTIFICATION_FUNCTION MyNotificationFunction;
+
+#define NTDLL_FUNCTION(name) reinterpret_cast<decltype(&name)>(GetProcAddress(ntdll, #name))
+
```

Comment:
Should these be inside the region?

Reply:
@dannyvv These are not defined by NT headers (the defs in the region are lifted from NT headers/MSDN)